src/MongoDB.Driver.Core/Core/Compression/Zstd/NativeBufferInfo.cs

```diff
@@ -0,0 +1,29 @@
+﻿/* Copyright 2019–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace MongoDB.Driver.Core.Compression.Zstd
+{
+    [StructLayout(LayoutKind.Sequential)]
+    internal class NativeBufferInfo
```

Comment:
I'm not sure, but if we really wanted to use a struct, we might be able to pass the struct by reference when calling the native code? I don't mind leaving it as a class though.

Reply:
it's possible to provide arguments by reference (via `ref`), but I don't think that adding ref options is a more good looking way here. It will require adding this modificator to any place where NativeBuffer is changed now.

Reply:
But if it was a struct it would have to be immutable. You are correct in pointing out that if we pass a struct to method we call and the method attempts to modify the struct it will modify a copy instead.